### PR TITLE
Fix formatter to work properly with function signatures

### LINKED ISSUE #37098: Formatting doesn't work properly with function signatures
**Description:**
Since, there is no style guidelines defined for function signatures, when the function signature has parameters of type like `type-descriptor` or `object-constructor` the formatting behaves unexpectedly as mentioned in the issue #34961

Since this is not specific to `object-constructors` as mentioned in the issue #34961. Some styling guideline is defined as below for function signatures

Example 1:

```ballerina
 function testFunction(int y,
        record {
            int z;
            int x;
        } rec) {
    // method body
}

```

Example 2:

```ballerina
function testFunction(int y, record {
            int z;
            int x;
        } rec) {
    // method body
}

```

### PR DESCRIPTION
**Pull request by AzeemMuzammil**
Fixes #37098, #34961

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by NipunaRanasinghe — APPROVED** [review]
LGTM

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java [hunk 1]
```diff
@@ -344,15 +344,17 @@ public ResourcePathParameterNode transform(ResourcePathParameterNode resourcePat
...
-        // start from the same level as the open parenthesis.
-        int currentIndentation = env.currentIndentation;
-        setIndentation(env.lineLength);
+        // Start a new indentation of two tabs for the parameters.
+        indent(2);
         SeparatedNodeList<ParameterNode> parameters =
-                formatSeparatedNodeList(functionSignatureNode.parameters(), 0, 0, 0, 0);
-        setIndentation(currentIndentation);
+                formatSeparatedNodeList(functionSignatureNode.parameters(), 0, 0, 0, 0, 0, 0, true);
+        unindent(2);
```
Comment (IrushiL) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java:357]: As discussed offline, let's create a separate issue to make the number of tabs configurable through formatting options.

Comment (AzeemMuzammil) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java:357]: Okay, lemme create that issue 👍

File: misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java [hunk 2]
```diff
@@ -73,6 +73,9 @@ public List<String> skipList() {
                 "qualified_identifier_assert_08.bal",
                 "conditional_expr_source_28.bal",
                 "resiliency_source_04.bal",
+                "record_type_def_source_14.bal",
+                "object_type_def_source_12.bal",
+                "anon_func_source_01.bal",
```
Comment (IrushiL) [inline misc/formatter/modules/formatter-core/src/test/java/org/ballerinalang/formatter/core/ParserTestFormatter.java:78]: In order to update parser test cases with correct formatting we can use the following steps:
- Change the formatting in the required parser test cases.
- Change the `UPDATE_ASSERTS` variable to `true`. https://github.com/ballerina-platform/ballerina-lang/blob/01f06c390aa875e0cc2134a74f8754d0c045e193/compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/ParserTestUtils.java#L75
- Run the parser test cases.
-  Change the `UPDATE_ASSERTS` variable back to `false`. 
- Commit the changed `.json` and `.bal` resource files.

When changing the parser test cases, make sure to get the PR approved by a parser team member as well.

**Review by IrushiL — APPROVED** [review]

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java (hunk 1 above)
Comment (AzeemMuzammil) [reply misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java:357]: #37552

**Pull request merged**